include/systems/elem_assembly.h

```diff
@@ -60,16 +61,17 @@ class ElemAssembly : public ReferenceCountedObject<ElemAssembly>
   virtual void boundary_assembly(FEMContext &) { }
 
   /**
-   * Get values to add to the RHS vector based on \p node.
-   * This allows one to impose point loads, for example.
+   * Get values to add to the matrix or rhs vector based on \p node.
+   * This allows one to impose point loads or springs, for example.
    */
   virtual void
-  get_nodal_rhs_values(std::map<numeric_index_type, Number> & values,
-                       const System &,
-                       const Node &)
+  get_nodal_values(std::vector<dof_id_type> ,
+                   DenseMatrix<Number>& ,
+                   DenseVector<Number>& ,
+                   const System & ,
+                   const Node &)
   {
-    // By default, just clear the values map
-    values.clear();
+    // Do nothing by default
   }
```

Comment:
This is API-breaking for a class that's intended to be subclassed, isn't it?  Any way to keep some (libmesh_deprecated()) backwards compatibility here?